Lib/test/test_typing.py

```diff
@@ -4091,6 +4091,20 @@ def method(self) -> None: ...
         self.assertIsInstance(Foo(), ProtocolWithMixedMembers)
         self.assertNotIsInstance(42, ProtocolWithMixedMembers)
 
+    def test_protocol_issubclass_error_message(self):
+        class Vec2D(Protocol):
+            x: float
+            y: float
+
+            def square_norm(self) -> float:
+                return self.x ** 2 + self.y ** 2
+
+        expected_error_message = (
+            "Protocols with non-method members don't support issubclass()."
+            " Non-method members: {'x', 'y'}."
+        )
+        self.assertRaisesRegex(TypeError, f'^{re.escape(expected_error_message)}$')
```

Comment:
You're not actually asserting anything here, you're just creating a context manager that you're immediately throwing away:

```pycon
>>> import unittest
>>> unittest.TestCase().assertRaisesRegex(TypeError, 'foo')
<unittest.case._AssertRaisesContext object at 0x11a9ecb80>
```

```suggestion
        with self.assertRaisesRegex(TypeError, re.escape(expected_error_message)):
            issubclass(int, Vec2D)

```